packages/remix-validated-form/src/hooks.ts

```diff
@@ -20,7 +20,7 @@ export type FieldProps = {
   /**
    * Validates the field.
    */
-  validate: () => void;
+  validate: () => Promise<void>;
```

Comment:
From the perspective of the `useField` consumer, `validate` should be fire-and-forget (i.e. `() => void`). No need to make it a promise.

Reply:
It's still fire-and-forget, the typing just lets you know that a promise is there if you want it right?

Reply:
My comment disappeared? :eyes:

I think it should be typed correctly with the promise, it doesn't mean the consumer has to care about it... but the validator will return a promise

Reply:
I worry that returning a promise will communicate that it _should_ be awaited, not that it can be. Let's keep the Promise in `validatedField` and use it to implement an `isValidating` state in `useField`. The implementation should look something like

```
validate: () => {
  setValidating(true);
  validateField(name).then(() => setValidating(false));
  // not returning the promise
}
```

Reply:
Hmmm you shouldnt hide a promise based on the off chance the Dev might "think" they have to await it (which doesn't matter anyways)

It should be an async function so the user *can* await it if they need to... Easy enough to mention it in the docs

Reply:
If we add an `isValidating` there's really no reason to ever await `validate`. Just because the implementation kicks off async logic doesn't mean the API should expose it. The function is fire-and-forget and the API should say that.

Reply:
Yeah I mean I can't think of any examples either... Would there be no scenarios you want some UI change after validation?

I just want to make sure this isnt coddling the Devs, to their detriment haha

Reply:
UI changes that could be implemented through awaiting the promise would be better done using `error` and `isValidating`. If there's really some edge case where you need it, there's still the promise on `valudateField` in the form context.

Actually, instead of `isValidating` we should go for `validatonState` which would be `'idle' | | 'validating' | 'valid' | 'invalid'`. That should cover everything.

Reply:
We could leave this out of this pr probably and just focus on the core async submission functionality.

Reply:
Yeah that sounds better